wizard/class-instant-articles-wizard-state.php

```diff
@@ -0,0 +1,352 @@
+	const TRANSITION_START_WIZARD = 'TRANSITION_START_WIZARD';
+	const TRANSITION_SET_UP_APP = 'TRANSITION_SET_UP_APP';
+	const TRANSITION_SELECT_PAGE = 'TRANSITION_SELECT_PAGE';
+	const TRANSITION_SELECT_STYLE = 'TRANSITION_SELECT_STYLE';
+	const TRANSITION_EDIT_STYLE = 'TRANSITION_EDIT_STYLE';
+	const TRANSITION_EDIT_PAGE = 'TRANSITION_EDIT_PAGE';
+	const TRANSITION_EDIT_APP = 'TRANSITION_EDIT_APP';
+
+	// WIZARD TIMELINE
+	const TIMELINE_PAST = 'TIMELINE_PAST';
+	const TIMELINE_CURRENT = 'TIMELINE_CURRENT';
+	const TIMELINE_FUTURE = 'TIMELINE_FUTURE';
+
+
+	/**
+	 * Transition vectors, format:
+	 *
+	 * array(
+	 *   ORIGINAL_STATE => array(
+	 *     NEW_STATE => TRANSITION_NAME,
+	 *     ...
+	 *   ),
+	 *   ...
+	 * )
+	 *
+	 */
+	public static $transition_vectors = array(
+		self::STATE_OVERVIEW => array (
+			self::STATE_APP_SETUP => self::TRANSITION_START_WIZARD
+		),
+		self::STATE_APP_SETUP => array (
+			self::STATE_PAGE_SELECTION => self::TRANSITION_SET_UP_APP
+		),
+		self::STATE_PAGE_SELECTION => array (
+			self::STATE_STYLE_SELECTION => self::TRANSITION_SELECT_PAGE,
+			self::STATE_APP_SETUP => self::TRANSITION_EDIT_APP
+		),
+		self::STATE_STYLE_SELECTION => array (
+			self::STATE_REVIEW_SUBMISSION => self::TRANSITION_SELECT_STYLE,
+			self::STATE_PAGE_SELECTION => self::TRANSITION_EDIT_PAGE,
+			self::STATE_APP_SETUP => self::TRANSITION_EDIT_APP
+		),
+		self::STATE_REVIEW_SUBMISSION => array (
+			self::STATE_STYLE_SELECTION => self::TRANSITION_EDIT_STYLE,
+			self::STATE_PAGE_SELECTION => self::TRANSITION_EDIT_PAGE,
+			self::STATE_APP_SETUP => self::TRANSITION_EDIT_APP
+		),
+	);
+
+	/**
+	 * Order of states on the wizard
+	 */
+	public static $timeline = array(
+		self::STATE_OVERVIEW => 0,
+		self::STATE_APP_SETUP => 1,
+		self::STATE_PAGE_SELECTION => 2,
+		self::STATE_STYLE_SELECTION => 3,
+		self::STATE_REVIEW_SUBMISSION => 4
+	);
+
+	/**
+	 * Gets the timeline position for a given state.
+	 *
+	 * @param string $state The state constant
+	 * @return string The timeline constant (PAST, CURRENT or FUTURE)
+	 */
+	public static function get_timeline_position( $state ) {
+		$current_state = self::get_current_state();
+
+		if ( self::$timeline[ $current_state ] > self::$timeline[ $state ] ) {
+			return self::TIMELINE_PAST;
+		} elseif ( $state === $current_state ) {
+			if (
+				$current_state === self::STATE_REVIEW_SUBMISSION &&
+				Instant_Articles_Wizard_Review_Submission::getReviewSubmissionStatus() === Instant_Articles_Wizard_Review_Submission::STATUS_APPROVED
+			) {
+				return self::TIMELINE_PAST;
+			}
+			return self::TIMELINE_CURRENT;
+		} else {
+			return self::TIMELINE_FUTURE;
+		}
+	}
+
+	/**
+	 * Retrieves the current state of the wizard
+	 *
+	 * @return string The state constant for the current state
+	 */
+	public static function get_current_state() {
+
+		$option = get_option( 'instant-articles-current-state', null );
+		if ( in_array( $option, array_keys( self::$transition_vectors ) ) ) {
```

Comment:
Ops! Actually `$option` is the needle not the haystack, so it does not need to be an array.